rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -85,23 +83,8 @@ def import_rmw_implementation():
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
     available_implementations = get_rmw_implementations()
-    if __selected_rmw_implementation is None:
-        logger = logging.getLogger('rclpy')
-        # prefer FastRTPS, otherwise first in alphabetical order
-        # the same logic is implemented in
-        # rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake
-        default_impl = 'rmw_fastrtps_cpp'
-        if default_impl in available_implementations:
-            select_rmw_implementation(default_impl)
-        elif available_implementations:
-            select_rmw_implementation(available_implementations[0])  # select the first one
-        else:
-            raise NoImplementationAvailableException()
-        logger.debug("Implicitly selecting the '{0}' rmw implementation."
-                     .format(__selected_rmw_implementation))
-    module_name = '._rclpy__{rmw_implementation}'.format(
-        rmw_implementation=__selected_rmw_implementation,
-    )
+    __selected_rmw_implementation = 'rmw_implementation'
```

Comment:
> I though[t] that splitting changes up into more bite size pieces is desired (for the sanity of the programmer making the patches as well as the sanity of the reviewer). 

I'm fine with breaking things up, but I would not split the work up as you have here. You could have left the one Python module per rmw impl design in place and only changed this repository to work with the new system, e.g. like change the default to `rmw_implementation` rather than "`rmw_fastrtps_cpp` or the first" as it is now. You could also have done what you've done here plus remove old code and tests, but what's here is, from my perspective, in between and not a good stopping place. That's all I'm saying.

> I will update this PR then to remove the "dead code" and disabled test as part of this PR since you prefer that.

I do prefer that, but others might disagree with me and approve it as-is.

> The `RMW_IMPLEMENTATION` environment variable works across all languages.

I'm not disputing that.

> In C++ you don't have the option to choose the rmw implementation from within the running executable.

Because it is not simple to do so technically, but that's not a good reason to not do it in Python, imo. There's lots of things you can easily do in Python that aren't practical in C++.

> Therefore I don't think this is a feature we need in Python - especially since it is not possible currently to use rmw A and then switch to rmw B within the same Python script.

I'm -0 on removing this feature. I think it would be potentially useful for tests (by using multiprocessing as some of our tests already do), despite the fact that you cannot use different ones from the same script at different times (without multiprocessing). But I understand not wanting to spend more time on fixing it for the new rmw mechanism.

> We should then make a decision on the question about selecting the rmw impl. within a running Python script in order to continue on this.

I'm slightly in favor of keeping it, mostly because I think it would be easy enough to fix no matter which route we take (leave the one Python module per rmw impl or try setting the env variable before importing the module with the rmw impl). I agree others should chime in on whether or not they want to preserve the feature.

If you wanted to work around this issue and merge these pr's sooner, simply change the default rmw implementation Python uses to be `rmw_implemenation`. I don't think anything else has to change. That's the least intrusive route in my opinion. Whether or not we do away with `RCLPY_IMPLEMENTATION` and avoid building multiple Python modules, one for each rmw impl, is a separate decision from my perspective.